cmd/kubelet/app/server.go

```diff
@@ -1151,7 +1159,8 @@ func startKubelet(k kubelet.Bootstrap, podCfg *config.PodConfig, kubeCfg *kubele
 
 	// start the kubelet server
 	if enableServer {
-		go k.ListenAndServe(net.ParseIP(kubeCfg.Address), uint(kubeCfg.Port), kubeDeps.TLSOptions, kubeDeps.Auth, enableCAdvisorJSONEndpoints, kubeCfg.EnableDebuggingHandlers, kubeCfg.EnableContentionProfiling)
+		go k.ListenAndServe(net.ParseIP(kubeCfg.Address), uint(kubeCfg.Port), kubeDeps.TLSOptions, kubeDeps.Auth,
+			enableCAdvisorJSONEndpoints, kubeCfg.EnableDebuggingHandlers, kubeCfg.EnableContentionProfiling, *kubeCfg.EnableSystemLogHandler)
```

Comment:
can `kubeCfg.EnableSystemLogHandler` be nil here?

Reply:
It shouldn't be given that before invoking Run() method inside NewKubeletCommand(), EnableSystemLogHandler is set if its nil. Please correct me if I'm wrong.

Reply:
Just digging deeper found that RunKubelet is also invoked from kubemark/hallow_kubelet which will not have the value set :/

Reply:
bingo :-)

I'd probably do the fallback to EnableDebuggingHandlers here, at the point of use.

Reply:
@liggitt I was thinking about adding a fallback here but instead I like this https://github.com/kubernetes/kubernetes/pull/87273/commits/b67081512da6b999dfc2bedcd957e66aa9b1bc61 for the following reasons
1. If we add a fallback here it has to stay forever (we cant eventually move this fallback to config defaults even after we completely deprecate --enable-debugging-handlers)
2. In future we can change hallow_node to set defaults similar to config/v1beta1/defaults.go so that it mimics kubelet behavior when some parameters are not set.
3. Potentially we would not expect any other consumers for RunKubelet() other than today's consumers right. 

Let me know what you think. Sorry it looked like simple PR but took lot of your time I guess :)

Reply:
I don't feel that strongly about putting the fallback here, but if this field being nil here can panic the kubelet, we should validate it is non-nil in ValidateKubeletConfiguration.

The way the kubelet config file and flags interact is pretty complicated (CLI flags can override config fields, cc @mtaufen), so I think the following paths are also possible:
* config file sets `enableDebuggingHandlers: false`, CLI flag sets `--enable-debugging-handlers=true`, we should serve logs
* config file sets `enableDebuggingHandlers: true`, CLI flag sets `--enable-debugging-handlers=false`, we should not serve logs

Reply:
Yes the code I have takes care of the cases mentioned above.

Reply:
Special casing a single field here is not ideal, and it's conventional to avoid pointers on the internal types. 

Alternatives:
- `DisableSystemLogHandler` instead, which can deny the `/logs` endpoint from being registered. Defaults to `false`, which has no effect, so unset is backwards compatible with either `true` or `false` for `EnableDebuggingHandlers`.
- Generic allow-list for debugging endpoints. Defaults to `["*"]`.
- Generic deny-list for debugging endpoints. Defaults to `[]`.